nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/ConfigParser.java

```diff
@@ -161,36 +182,55 @@ public DomDocument parse(URL source, DomDocument document, Dom parent) {
      */
-    protected Dom handleElement(XMLStreamReader in,DomDocument document, Dom parent) throws XMLStreamException {
+    protected Dom handleElement(XMLStreamReader in, DomDocument document, Dom parent) throws XMLStreamException {
         ConfigModel model = document.getModelByElementName(in.getLocalName());
         if(model==null) {
+            // Get the element name
             String localName = in.getLocalName();
-            Logger.getAnonymousLogger().severe("Ignoring unrecognized element "+in.getLocalName() + " at " + in.getLocation());
+
+            log(SEVERE, "Ignoring unrecognized element %s at %s", in.getLocalName(), in.getLocation());
+
             // flush the sub element content from the parser
             int depth=1;
             while(depth>0) {
                 final int tag = in.nextTag();
                 if (tag==START_ELEMENT && in.getLocalName().equals(localName)) {
-                    if (Logger.getAnonymousLogger().isLoggable(Level.FINE)) {
-                        Logger.getAnonymousLogger().fine("Found child of same type "+localName+" ignoring too");
-                    }
+                    log(FINE, "Found child of same type %s ignoring too", localName);
                     depth++;
                 }
                 if (tag==END_ELEMENT && in.getLocalName().equals(localName)) {
-                    if (Logger.getAnonymousLogger().isLoggable(Level.FINE)) {
-                        Logger.getAnonymousLogger().fine("closing element type " + localName);
-                    }
+                    log(FINE, "Closing element type %s", localName);
                     depth--;
                 }
-                if (Logger.getAnonymousLogger().isLoggable(Level.FINE) && tag==START_ELEMENT) {
-                    Logger.getAnonymousLogger().fine("Jumping over " + in.getLocalName());
+                if (tag == START_ELEMENT) {
+                    log(FINE, "Jumping over %s", in.getLocalName());
                 }
             }
             return null;
         }
         return handleElement(in,document,parent,model);
     }
 
+    /**
+     * If the provided level is not provided, this method will do nothing.
+     * Otherwise, it will format the message and add it to {@link #getErrors()}. If
+     * {@link #logErrors} is true, this method will also log the message.
+     * 
+     * @param level          the level to log at
+     * @param baseLogMessage the message to be passed to {@see String#format(String,
+     *                       Object...)}
+     * @param parameters     the parameters to be filled into the base log message
+     */
+    private void log(Level level, String baseLogMessage, Object... parameters) {
+        if (level != null && LOGGER.isLoggable(level)) {
+            String logMessage = String.format(baseLogMessage, parameters);
+            if (logErrors) {
+                LOGGER.log(level, logMessage);
+            }
+            this.errors.add(logMessage);
```

Comment:
GOTO LABEL1: so if I would set the logging level to ALL, this would add even FINE messages to errors.